diff --git a/utils/common/codec-fwht.h b/utils/common/codec-fwht.h
index cdfad133..51fd147c 100644
--- a/utils/common/codec-fwht.h
+++ b/utils/common/codec-fwht.h
@@ -8,8 +8,20 @@
 #define CODEC_FWHT_H
 
 #include <linux/types.h>
-#include <linux/bitops.h>
-#include <asm/byteorder.h>
+#include <string.h>
+#include <errno.h>
+#include <arpa/inet.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+#define BIT(x) (1 << (x))
+#define ARRAY_SIZE(x) (sizeof(x) / sizeof((x)[0]))
+
+typedef __u32 u32;
+typedef __u16 u16;
+typedef __s16 s16;
+typedef __s32 s32;
+typedef __u8 u8;
 
 /*
  * The compressed format consists of a fwht_cframe_hdr struct followed by the
