--- utils/common/v4l2-tpg.h.bak	2016-04-22 09:22:11.168936563 +0200
+++ utils/common/v4l2-tpg.h	2016-04-22 09:23:07.337262639 +0200
@@ -20,13 +20,56 @@
 #ifndef _V4L2_TPG_H_
 #define _V4L2_TPG_H_
 
-#include <linux/types.h>
-#include <linux/errno.h>
-#include <linux/random.h>
-#include <linux/slab.h>
-#include <linux/vmalloc.h>
 #include <linux/videodev2.h>
-#include <media/v4l2-tpg-colors.h>
+
+#include <stdbool.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+
+typedef __u32 u32;
+typedef __u16 u16;
+typedef __s16 s16;
+typedef __u8 u8;
+typedef __s8 s8;
+
+#define __packed __attribute__((packed))
+#define pr_info printf
+#define noinline
+
+#ifndef min
+#define min(a,b)	((a) < (b) ? (a) : (b))
+#define max(a,b)	((a) > (b) ? (a) : (b))
+#endif /* !min */
+#define min3(x, y, z) min((typeof(x))min(x, y), z)
+#define max3(x, y, z) max((typeof(x))max(x, y), z)
+
+static inline void vfree(void *p)
+{
+	free(p);
+}
+
+static inline void *vzalloc(unsigned long size)
+{
+	return calloc(1, size);
+}
+
+#define clamp(val, min, max) ({			\
+	typeof(val) __val = (val);		\
+	typeof(min) __min = (min);		\
+	typeof(max) __max = (max);		\
+	(void) (&__val == &__min);		\
+	(void) (&__val == &__max);		\
+	__val = __val < __min ? __min: __val;	\
+	__val > __max ? __max: __val; })
+
+static inline u32 prandom_u32_max(u32 ep_ro)
+{
+	return rand() % ep_ro;
+}
+
+#include "v4l2-tpg-colors.h"
 
 enum tpg_pattern {
 	TPG_PAT_75_COLORBAR,
--- ../media-git/drivers/media/common/v4l2-tpg/v4l2-tpg-core.c	2016-04-21 08:15:35.439610205 +0200
+++ utils/common/v4l2-tpg-core.c	2016-04-22 09:29:50.955939090 +0200
@@ -20,8 +20,7 @@
  * SOFTWARE.
  */
 
-#include <linux/module.h>
-#include <media/v4l2-tpg.h>
+#include "v4l2-tpg.h"
 
 /* Must remain in sync with enum tpg_pattern */
 const char * const tpg_pattern_strings[] = {
@@ -49,7 +48,6 @@
 	"Noise",
 	NULL
 };
-EXPORT_SYMBOL_GPL(tpg_pattern_strings);
 
 /* Must remain in sync with enum tpg_aspect */
 const char * const tpg_aspect_strings[] = {
@@ -60,7 +58,6 @@
 	"16x9 Anamorphic",
 	NULL
 };
-EXPORT_SYMBOL_GPL(tpg_aspect_strings);
 
 /*
  * Sine table: sin[0] = 127 * sin(-180 degrees)
@@ -96,7 +93,6 @@
 {
 	font8x16 = f;
 }
-EXPORT_SYMBOL_GPL(tpg_set_font);
 
 void tpg_init(struct tpg_data *tpg, unsigned w, unsigned h)
 {
@@ -118,7 +114,6 @@
 	tpg->colorspace = V4L2_COLORSPACE_SRGB;
 	tpg->perc_fill = 100;
 }
-EXPORT_SYMBOL_GPL(tpg_init);
 
 int tpg_alloc(struct tpg_data *tpg, unsigned max_w)
 {
@@ -155,7 +150,6 @@
 	}
 	return 0;
 }
-EXPORT_SYMBOL_GPL(tpg_alloc);
 
 void tpg_free(struct tpg_data *tpg)
 {
@@ -180,7 +174,6 @@
 		tpg->random_line[plane] = NULL;
 	}
 }
-EXPORT_SYMBOL_GPL(tpg_free);
 
 bool tpg_s_fourcc(struct tpg_data *tpg, u32 fourcc)
 {
@@ -410,7 +403,6 @@
 	}
 	return true;
 }
-EXPORT_SYMBOL_GPL(tpg_s_fourcc);
 
 void tpg_s_crop_compose(struct tpg_data *tpg, const struct v4l2_rect *crop,
 		const struct v4l2_rect *compose)
@@ -426,7 +418,6 @@
 		tpg->scaled_width = 2;
 	tpg->recalc_lines = true;
 }
-EXPORT_SYMBOL_GPL(tpg_s_crop_compose);
 
 void tpg_reset_source(struct tpg_data *tpg, unsigned width, unsigned height,
 		       u32 field)
@@ -451,7 +442,6 @@
 				       (2 * tpg->hdownsampling[p]);
 	tpg->recalc_square_border = true;
 }
-EXPORT_SYMBOL_GPL(tpg_reset_source);
 
 static enum tpg_color tpg_get_textbg_color(struct tpg_data *tpg)
 {
@@ -1260,7 +1250,6 @@
 		return 0;
 	}
 }
-EXPORT_SYMBOL_GPL(tpg_g_interleaved_plane);
 
 /* Return how many pattern lines are used by the current pattern. */
 static unsigned tpg_get_pat_lines(const struct tpg_data *tpg)
@@ -1736,7 +1725,6 @@
 		}
 	}
 }
-EXPORT_SYMBOL_GPL(tpg_gen_text);
 
 void tpg_update_mv_step(struct tpg_data *tpg)
 {
@@ -1785,7 +1773,6 @@
 	if (factor < 0)
 		tpg->mv_vert_step = tpg->src_height - tpg->mv_vert_step;
 }
-EXPORT_SYMBOL_GPL(tpg_update_mv_step);
 
 /* Map the line number relative to the crop rectangle to a frame line number */
 static unsigned tpg_calc_frameline(const struct tpg_data *tpg, unsigned src_y,
@@ -1875,7 +1862,6 @@
 	if (p == 0 && tpg->interleaved)
 		tpg_calc_text_basep(tpg, basep, 1, vbuf);
 }
-EXPORT_SYMBOL_GPL(tpg_calc_text_basep);
 
 static int tpg_pattern_avg(const struct tpg_data *tpg,
 			   unsigned pat1, unsigned pat2)
@@ -1905,7 +1891,6 @@
 	pr_info("tpg quantization: %d/%d\n", tpg->quantization, tpg->real_quantization);
 	pr_info("tpg RGB range: %d/%d\n", tpg->rgb_range, tpg->real_rgb_range);
 }
-EXPORT_SYMBOL_GPL(tpg_log_status);
 
 /*
  * This struct contains common parameters used by both the drawing of the
@@ -2311,7 +2296,6 @@
 				vbuf + buf_line * params.stride);
 	}
 }
-EXPORT_SYMBOL_GPL(tpg_fill_plane_buffer);
 
 void tpg_fillbuffer(struct tpg_data *tpg, v4l2_std_id std, unsigned p, u8 *vbuf)
 {
@@ -2328,8 +2312,3 @@
 		offset += tpg_calc_plane_size(tpg, i);
 	}
 }
-EXPORT_SYMBOL_GPL(tpg_fillbuffer);
-
-MODULE_DESCRIPTION("V4L2 Test Pattern Generator");
-MODULE_AUTHOR("Hans Verkuil");
-MODULE_LICENSE("GPL");
--- ../media-git/drivers/media/common/v4l2-tpg/v4l2-tpg-colors.c	2016-04-21 08:15:35.439610205 +0200
+++ utils/common/v4l2-tpg-colors.c	2016-04-22 09:26:24.638509385 +0200
@@ -36,7 +36,7 @@
  */
 
 #include <linux/videodev2.h>
-#include <media/v4l2-tpg-colors.h>
+#include "v4l2-tpg-colors.h"
 
 /* sRGB colors with range [0-255] */
 const struct color tpg_colors[TPG_COLOR_MAX] = {
